### PR TITLE
330 ingredient dynamic search

### PR DESCRIPTION
**Pull request by MohiuddinRU**
# Description

Refactored according to PR reviews

Fixes # (issue)

## Type of change

Adding error handling to some methods.
Naming conventions etc.
Pulling from `develop` branch and resolved a merge conflict.

# How Has This Been Tested?
configuration

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/server/prisma/seeder.ts
Comment (MNSM92) [inline apps/server/prisma/seeder.ts:1]: Is this related to the token? If we change any other change rather than the token, we need another token.

File: apps/server/prisma/seed.ts
Comment (MNSM92) [inline apps/server/prisma/seed.ts:1]: Is this related to the token? If we change any other change rather than the token, we need another token.

File: apps/server/prisma/ingredients.json
Comment (MNSM92) [inline apps/server/prisma/ingredients.json:1]: is "ingredients" wrong?

File: apps/server/models/IngredientModel.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+      if (!ingredient) return null
+
+      const { name, imageName } = ingredient
+
+      return {
+        id: ingredient.id,
+        name,
+        imageName
+      }
```
Comment (MNSM92) [inline apps/server/models/IngredientModel.ts]: what if?

      const { id, name, imageName } = ingredient

      return {
        id,
        name,
        imageName
      }

File: apps/server/controllers/ingredient/ingredient.controller.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      const { id } = req.params
+      const ingredient = await super.read(id)
+      res.status(200).json(ingredient)
+    } catch (error) {
+      console.log(error)
+      res.status(500).json('Internal Server Error')
+    }
+  }
+
+  async getAllIngredients(req: Request, res: Response) {
```
Comment (MNSM92) [inline apps/server/controllers/ingredient/ingredient.controller.ts:16]: is req: Request needed here?

File: apps/feedahelp/src/components/ui/features/ingredients/Ingredients.tsx [hunk 3]
```diff
@@ -4,20 +4,34 @@ import { fetchIngredients } from "~/utils/cms/fetchIngredients";
...
 const Ingredients = () => {
-  const [ingredients, setIngredients] = useState([]);
+  const [ingredients, setIngredients] = useState<any[]>([]);
+  const [searchedIngredients, setSearchedIngredients] = useState<any[]>([]);
 
   const getIngredients = async () => {
     try {
-      const socialData = await fetchIngredients();
-      setIngredients(socialData);
+      const { data } = await getMethod("ingredient/all", "");
```
Comment (MNSM92) [inline apps/feedahelp/src/components/ui/features/ingredients/Ingredients.tsx]: there is no { data } in response of api

File: apps/feedahelp/src/components/ui/features/ingredients/Ingredients.tsx [hunk 4]
```diff
@@ -28,22 +42,27 @@ const Ingredients = () => {
...
                   className="relative flex cursor-pointer justify-center rounded-full"
                 >
                   <IngredientCheckbox
-                    id={ingredient.englishName}
-                    imgSrc={urlForThumbnail(ingredient.image)}
-                    imgAlt={ingredient.englishName}
+                    id={ingredient.id}
+                    //todo: need to decide how to resolve this
+                    //imgSrc={urlForThumbnail(ingredient.imageName)}
+                    imgSrc=""
```
Comment (MNSM92) [inline apps/feedahelp/src/components/ui/features/ingredients/Ingredients.tsx]: imgSrc???

**Review by MNSM92 — COMMENTED** [review]
LGTM, few questions

File: apps/server/prisma/seed.ts
Comment (MohiuddinRU) [reply apps/server/prisma/seed.ts:1]: Couldn't understand. Could you please elaborate your question?

File: apps/server/prisma/ingredients.json
Comment (MohiuddinRU) [reply apps/server/prisma/ingredients.json:1]: No, it's alright. I have tried my level best to retain the name. But, the code will not work if we don't change here. Because, it's related to the model name. You check changing the name.

File: apps/server/models/IngredientModel.ts (hunk 1 above)
Comment (MohiuddinRU) [reply apps/server/models/IngredientModel.ts]: Yes, possible. Thanks.
